core/src/main/java/hudson/model/Project.java

```diff
@@ -107,6 +108,11 @@ public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOExcep
     @Override public Item asItem() {
         return this;
     }
+    
+    @Exported
+    public String getName() {
+        return Item.getName();
```

Comment:
Not a useful response as soon as folders are involved.

Reply:
Project is inheriting from AbstractItem that already exports getName which is the implementation of Item.getName. So this is completely nonsense. That code isn't compiling